common-npm-packages/packaging-common/nuget/ProductVersionHelper.ts

```diff
@@ -0,0 +1,12 @@
+import {VersionInfo} from "../pe-parser/VersionResource";
+
+export function getVersionFallback(version: VersionInfo): VersionInfo {
+    const productVersion = version.productVersion;
+    if ((productVersion.a === 0) && (productVersion.b === 0) && (productVersion.c === 0) && (productVersion.d === 0)) {
+        version.productVersion.a = version.fileVersion.a;
+        version.productVersion.b = version.fileVersion.b;
+        version.productVersion.c = version.fileVersion.c;
+        version.productVersion.d = version.fileVersion.d;
+    }
+    return version;
```

Comment:
Instead of altering the passed-in version, you could return the whole `version.productVersion` or `version.fileVersion` object. Generally, callers don't expect functions to modify their arguments, especially when the function's name starts with 'get'.